cmd/ak/cmd/sessions/start.go

```diff
@@ -15,70 +14,53 @@ import (
 )
 
 const (
-	defaultWait = 1 * time.Second
+	defaultPollInterval = 1 * time.Second
 )
 
 var (
-	entryPoint   string
-	memos        []string
-	wait         bool
-	waitInterval time.Duration
+	entryPoint string
+	memos      []string
+	build      string
+	deployID   string
 )
 
 var startCmd = common.StandardCommand(&cobra.Command{
-	Use:   "start <--deployment-id=...> <--event-id=...> <--entrypoint=...> [--memo=...] [--wait]",
+	Use:   "start [--build-id=...] [--env=...] [--deployment-id=...] <--entrypoint=...> [--memo=...] [--watch] [--watch-timeout=...] [--poll-interval=...] [--no-timestamps] [--quiet]",
 	Short: "Start new session",
 	Args:  cobra.NoArgs,
 
 	RunE: func(cmd *cobra.Command, args []string) error {
-		r := resolver.Resolver{Client: common.Client()}
-		d, did, err := r.DeploymentID(deploymentID)
-		if err != nil {
-			return err
+		if deployID == "" && build == "" {
+			return errors.New("either --deployment-id or --build-id must be provided")
 		}
-		if !d.IsValid() {
-			err = fmt.Errorf("deployment ID %q not found", deploymentID)
-			return common.NewExitCodeError(common.NotFoundExitCode, err)
+
+		if deployID != "" && (build != "" || env != "") {
+			return errors.New("--deployment-id cannot be used with --build-id or --env")
 		}
 
-		e, eid, err := r.EventID(eventID)
+		did, eid, bid, ep, err := sessionArgs()
 		if err != nil {
 			return err
 		}
-		if !e.IsValid() {
-			err = fmt.Errorf("event ID %q not found", eventID)
-			return common.NewExitCodeError(common.NotFoundExitCode, err)
-		}
 
-		ep, err := sdktypes.StrictParseCodeLocation(entryPoint)
-		if err != nil {
-			return fmt.Errorf("invalid entry-point %q: %w", entryPoint, err)
+		if !ep.IsValid() {
```

Comment:
maybe to move this check in sessionsArgs as well?

Reply:
it doesn't have to be specified in other use of sessionArgs.